### PR TITLE
util: Add SetupHelpOptions()

### PR DESCRIPTION
**Pull request by maflcko**
Every binary we have sets up the help option in their own way and wording.

Solve that by having one function take care of it for all of them.

### CODE REVIEW

File: src/util/system.cpp [hunk 1]
```diff
@@ -635,6 +635,12 @@ bool HelpRequested(const ArgsManager& args)
     return args.IsArgSet("-?") || args.IsArgSet("-h") || args.IsArgSet("-help") || args.IsArgSet("-help-debug");
 }
 
+void SetupHelpOptions(ArgsManager& args)
+{
+    args.AddArg("-?", "Print this help message and exit", false, OptionsCategory::OPTIONS);
```
Comment (promag) [inline src/util/system.cpp]: Have you considered doing this on `ArgsManager` constructor?

**Review by promag — COMMENTED** [review]
Concept ACK.

File: src/util/system.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/util/system.cpp]: I'd prefer not to bloat the constructor with stuff, since we've had countless initialization and memory issues due to that.

Comment (promag) [reply src/util/system.cpp]: Fair, you can resolve this.

File: src/util/system.h [hunk 2]
```diff
@@ -295,6 +295,9 @@ extern ArgsManager gArgs;
  */
 bool HelpRequested(const ArgsManager& args);
 
+/** Add help options to the args manager */
+void SetupHelpOptions(ArgsManager& args);
```
Comment (promag) [inline src/util/system.h]: nit, how this as a member, like `ArgsManager::SetupHelpOptions`?

**Comment by practicalswift** [comment]
Concept ACK

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15358/commits/a99999cc04c2e7d79bb841fd954c84e985ed3b88

File: src/util/system.h (hunk 2 above)
Comment (laanwj) [reply src/util/system.h]: Agree with @MarcoFalke here. It is not the responsibility of `ArgsManager` to know about or hardcode specific options.

**Comment by laanwj** [comment]
utACK a99999cc04c2e7d79bb841fd954c84e985ed3b88

**Pull request merged**